interfaces/prompting/requestrules/requestrules.go

```diff
@@ -507,37 +490,41 @@ func (rdb *RuleDB) addRulePermissionToTree(rule *Rule, permission string) []prom
 		return conflicts
 	}
 
-	for ruleID := range expiredRules {
-		removedRule, err := rdb.removeRuleByID(ruleID)
+	expiredData := map[string]string{"removed": "expired"}
+	for ruleID := range partiallyExpiredRules {
+		maybeExpired, err := rdb.lookupRuleByIDForUser(rule.User, ruleID)
+		if err != nil {
+			// Error shouldn't occur. If it does, the rule was already removed
+			continue
+		}
+		// Already removed the rule's permission from the tree, let's remove
+		// it from the rule as well
+		delete(maybeExpired.Constraints.Permissions, permission)
```

Comment:
is it relevant to do this delete if we hit the code below that removes the entire rule? shouldn't we do this only in the non-expired case?

Reply:
The end result is the same (since pruning an expired permission from a rule which is not fully expired does not result in a notice), but might as well only do this if `!maybeExpired.expired()`. I've made the change.